contrib/cirrus/cache_release_archive.sh

```diff
@@ -0,0 +1,137 @@
+    if [[ "$SPECIALMODE" == "windows" ]] || [[ "$SPECIALMODE" == "darwin" ]]
+    then
+        RELFILE="podman-remote-${SPECIALMODE}.zip"
+    elif [[ "$SPECIALMODE" == "none" ]]
+    then
+        RELFILE="podman.zip"
+    else
+        die 55 "$(basename $0) unable to handle \$SPECIALMODE=$SPECIALMODE for $ARCHIVE_NAME"
+    fi
+    echo "Calling make $RELFILE"
+    cd $GOSRC
+    make "$RELFILE"
+    echo "Renaming archive so it can be identified/downloaded for publishing"
+    mv -v "$RELFILE" "$ARCHIVE_NAME"
+    echo "Success!"
+}
+
+[[ "$CI" == "true" ]] || \
+    die 56 "$0 requires a Cirrus-CI cross-task cache to function"
+
+cd $GOSRC
+# Same script re-used for both uploading and downloading to avoid duplication
+if [[ "$(basename $0)" == "cache_release_archive.sh" ]]
+then
+    # ref: https://cirrus-ci.org/guide/writing-tasks/#environment-variables
+    req_env_var CI_NODE_INDEX CIRRUS_BUILD_ID
+    # Use unique names for uncache_release_archives.sh to find/download them all
+    ARCHIVE_NAME="build-${CIRRUS_BUILD_ID}-task-${CI_NODE_INDEX}.zip"
+    make_release "$ARCHIVE_NAME"
+
+    # ref: https://cirrus-ci.org/guide/writing-tasks/#http-cache
+    URL="http://$CIRRUS_HTTP_CACHE_HOST/${ARCHIVE_NAME}"
+    echo "Uploading $ARCHIVE_NAME to Cirrus-CI cache at $URL"
+    curl -s -X POST --data-binary "@$ARCHIVE_NAME" "$URL"
+elif [[ "$(basename $0)" == "uncache_release_archives.sh" ]]
+then
+    req_env_var CIRRUS_BUILD_ID CI_NODE_TOTAL GCPJSON GCPNAME GCPROJECT
+    [[ "${CI_NODE_INDEX}" -eq  "$[CI_NODE_TOTAL-1]" ]] || \
+        die 8 "The release task must be executed last to guarantee archive cache is complete"
+
+    if [[ -n "$CIRRUS_PR" ]]
+    then
+        PR_OR_BRANCH="pr$CIRRUS_PR"
+        BUCKET="libpod-pr-releases"
+    elif [[ -n "$CIRRUS_BRANCH" ]]
+    then
+        PR_OR_BRANCH="$CIRRUS_BRANCH"
+        BUCKET="libpod-$CIRRUS_BRANCH-releases"
+    else
+        die 10 "Expecting either \$CIRRUS_PR or \$CIRRUS_BRANCH to be non-empty."
+    fi
+
+    echo "Blindly downloading Cirrus-CI cache files for task (some will fail)."
+    set +e  # Don't stop looping until all task's cache is attempted
+    for (( task_number = 0 ; task_number < $CI_NODE_TOTAL ; task_number++ ))
+    do
+        ARCHIVE_NAME="build-${CIRRUS_BUILD_ID}-task-${task_number}.zip"
+        URL="http://$CIRRUS_HTTP_CACHE_HOST/${ARCHIVE_NAME}"
+        echo "Attempting to download cached archive from $URL"
+        handle_archive "$task_number" "$PR_OR_BRANCH" "$URL"
+        echo "----------------------------------------"
+    done
+    set -e
+
+    [[ -n "$RELEASE_ARCHIVE_NAMES" ]] || \
+        die 67 "Error: No release archives found in CI cache, expecting at least one."
+
+    echo "Preparing to upload release archives."
+    gcloud config set project "$GCPROJECT"
+    echo "$GCPJSON" > /tmp/gcp.json
+    gcloud auth activate-service-account --key-file=/tmp/gcp.json || rm /tmp/gcp.json
+    gsutil -m cp /*.zip "gs://$BUCKET"  # Upload in parallel
```

Comment:
The `_task` suffix is consumed by Cirrus-CI, so the actual name shows up as 'release' in the WebUI.  Here's the most recent one: https://cirrus-ci.com/task/4874931624476672

Yes, it is because of line 47.  This is somewhat an artifact of a prior implementation which required the files to be in the root directory.  However, I kept it because the upload is much faster than doing them one-by-one in `handle_archive()` and testing-laziness.

Reply:
I _was_ looking without `_task`, and could swear that the string `release` did not appear in the `Checks` tab earlier today. I see it now, and it looks good, thanks.

Reply:
okay good, thx for looking.